src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -616,7 +615,10 @@ public void ContinuationAction(Task<bool> targetTask)
                 else
                 {
                     // completed successfully
-                    JobTcs.TrySetResult(targetTask.Result);
+                    jtf.Run(async () =>
+                    {
+                        JobTcs.TrySetResult(await targetTask);
+                    });
                 }
```

Comment:
This is an equivalent change, but the goal of the code looks like it could easily be misused. I see from this method's one caller that `targetTask` is already completed, so the await will no-op. But if this method were ever passed an incomplete task, your "continuation" would in fact execute inline and block while waiting. 
Given your assumption seems to be the task is already completed, you can simplify this by removing the jtf use, and just using `Assumes.True(targetTask.IsCompleted)` at the top of your method, then go ahead and use `targetTask.Result` as you were before.
Alternatively if you'd like to allow for incomplete tasks as the first argument, you can schedule the continuation so that it doesn't block while waiting.

Reply:
The analyzer's insight into whether the task is complete is rudimentary. Given a `Task` instance in a local variable, we can't tell if it's complete at compile-time or not. Roslyn doesn't give us flow analysis APIs yet where the analyzer could see that you had done a check that `Task.IsCompleted` either.
What the latest versions of the analyzer *can* do is recognize when the `task` local comes from a `ContinueWith` delegate parameter, such that we can assume it is in fact completed. But that isn't the case here -- nothing in this method provides any guarantee that `targetTask` is completed, so the analyzer is correct in that you shouldn't simply call `.Result` on it. Now, if you use `Assumes.True(targetTask.IsCompleted)`, that makes it clear (to the reader) that `.Result` is safe, but the analyzer isn't that smart. So yes, you'd need to suppress the warning at that point.